src/multi-select/tp-multi-select-search.ts

```diff
@@ -14,15 +14,34 @@ export class TPMultiSelectSearchElement extends HTMLElement {
 	 */
 	connectedCallback(): void {
 		const input: HTMLInputElement | null = this.querySelector( 'input' );
-		if ( ! input ) {
+		const multiSelect: TPMultiSelectElement | null = this.closest( 'tp-multi-select' );
+		if ( ! input || ! multiSelect ) {
 			return;
 		}
 
 		input.addEventListener( 'keydown', this.handleKeyboardInputs.bind( this ) );
 		input.addEventListener( 'keyup', this.handleSearchChange.bind( this ) );
 		input.addEventListener( 'input', this.handleSearchChange.bind( this ) );
 		this.addEventListener( 'click', this.handleClick.bind( this ) );
-		this.closest( 'tp-multi-select' )?.addEventListener( 'open', this.focus.bind( this ) );
+		multiSelect?.addEventListener( 'open', this.focus.bind( this ) );
+		multiSelect?.addEventListener( 'select-all', () => this.toggleSearchInputVisibility( false ) );
```

Comment:
@imranhsayed Not sure about this approach. Lets add an attribute instead and use CSS instead?

Reply:
Could you elaborate @junaidbhura ? We are using attribute called 'active' inside `toggleSearchInputVisibility` function?

Reply:
@imranhsayed Maybe we can use an attribute called `select-all="yes"` when its selected all on the parent? That way we can use it to make any special styling as well?

So basically listen for an attribute changed callback `select-all="yes"` and trigger this function: https://github.com/Travelopia/web-components/blob/master/src/multi-select/tp-multi-select.ts#L263

And if its removed, trigger this function: https://github.com/Travelopia/web-components/blob/master/src/multi-select/tp-multi-select.ts#L289

Then, here, just toggle the attribute instead of calling the function: https://github.com/Travelopia/web-components/blob/master/src/multi-select/tp-multi-select-select-all.ts#L48